test/e2e_node/services/kubelet.go

```diff
@@ -198,7 +200,20 @@ func (e *E2EServices) startKubelet() (*server, error) {
 		// Since kubelet will typically be run as a service it also makes more
 		// sense to test it that way
 		isSystemd = true
-		unitName := fmt.Sprintf("kubelet-%d.service", rand.Int31())
+		// To create the systemd unit name, try to find the timestamp
+		// used in the current directory, if it is named /tmp/node-e2e-<timestamp>
+		// then we can match it.
+		//
+		// If we can't find that timestamp, default to using Now()
+		unitTimestamp := fmt.Sprintf(time.Now().Format(timestampFormat))
+		if cwd, err := os.Getwd(); err == nil {
+			dirTimestamp := strings.TrimPrefix(filepath.Base(cwd), "node-e2e-")
```

Comment:
I do agree that using timestamp is much better than the random id. It is really hard for me to find the corresponding kubelet service.

However, it is not ideal that this is replying on an implementation detail in `remote` package https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/remote/remote.go#L69.

The directory name may change in the future, and this logic may break without awareness...

Can we add helper function in `remote` package, and use it here?

Reply:
Indeed, much better to have this in remote.go together with the other code for creation of the workspace...

I pushed an update. Retested and it still works as expected. PTAL.